claims_hosp/HospClaims/automate/sanity_checks.py

```diff
@@ -0,0 +1,289 @@
+        data["epidata_date_range"] = Epidata.range(
+            str(data["first_plot_date"].date()).replace('-', ''),
+            str(data["last_date"].date()).replace('-', ''))
+
+        return data
+
+    def check_se_na(self):
+        """
+        Checks that all SE are reported as 'NA' due to
+        privacy concerns from the company.
+
+        Returns:
+          true if pass, false otherwise
+        """
+
+        for kind in ["adj", "nadj"]:
+            for geo, ses in self.data[kind]["se"].items():
+                for se in ses:
+                    if not np.isnan(se):
+                        logging.error(f"{geo}, {se} not nan")
+                        return False
+        return True
+
+    def check_range(self):
+        """
+        Checks that all percentages are within [0, 100].
+
+        Returns:
+          true if pass, false otherwise
+        """
+        for kind in ["adj", "nadj"]:
+            for geo, vals in self.data[kind]["val"].items():
+                for val in vals:
+                    if not (0 <= val <= 100):
+                        logging.error(f"{geo}, {val} not in [0, 100]")
+                        return False
+        return True
+
+    def check_quantity(self):
+        """Checks how many geographies were generated."""
+        n_geos = {}
+        logging.info(f"geographies generated for {self.level}")
+        for kind in ["adj", "nadj"]:
+            for geo, vals in self.data[kind]['val'].items():
+                n_geos[geo] = len(vals)
+
+            min_geo = np.min([v for k, v in n_geos.items()])
+            max_geo = np.max([v for k, v in n_geos.items()])
+            avg_geo = np.mean([v for k, v in n_geos.items()])
+            std_geo = np.std([v for k, v in n_geos.items()])
+            logging.info(f"\t{kind}"
+                         f"\nmin:\t{min_geo}\nmax:\t{max_geo}"
+                         f"\navg:\t{avg_geo:.2f}\nstd:\t{std_geo:.2f}")
+
+    def get_filled_df(self, loc, kind):
+        df = pd.DataFrame({"val": self.data[kind]["val"][loc]},
+                          index=pd.to_datetime(self.data[kind]["dates"][loc]))
+
+        if self.data["first_plot_date"] not in df.index:
+            df = df.append(
+                pd.DataFrame({"val": np.nan}, index=[self.data["first_plot_date"]]))
+        if self.data["last_date"] not in df.index:
+            df = df.append(pd.DataFrame({"val": np.nan}, index=[self.data["last_date"]]))
+        df.sort_index(inplace=True)
+        df = df.asfreq('D', fill_value=np.nan)
+        return df[df.index > self.data["first_plot_date"]]
+
+    def get_epidata_df(self, loc, kind):
+        epi_kind = "smoothed_adj_covid19_from_claims" if kind == "adj" else "smoothed_covid19_from_claims"
+        if self.level == "msa":
+            loc = int(float(loc))
+
+        rows = Epidata.covidcast("hospital-admissions", epi_kind, "day",
+                                 self.level, self.data["epidata_date_range"], loc)
+        vals = []
```

Comment:
When the server is cranky, this will fail weirdly later on; check the response first instead
```suggestion
        Epidata.check(rows)
        vals = []
```